cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/MultipleOutputs.java

```diff
@@ -119,13 +122,23 @@ private static void checkNamedOutputName(String namedOutput, Collection<String>
   @SuppressWarnings("unchecked")
   public static void addNamedOutput(Job job, String namedOutput, String outputFormatClass,
                                     Class<?> keyClass, Class<?> valueClass, Map<String, String> outputConfigs) {
+    assertAlphaNumeric(namedOutput);
     checkNamedOutputName(namedOutput, getNamedOutputsList(job), false);
     Configuration conf = job.getConfiguration();
     conf.set(MULTIPLE_OUTPUTS, conf.get(MULTIPLE_OUTPUTS, "") + " " + namedOutput);
     conf.set(MO_PREFIX + namedOutput + FORMAT, outputFormatClass);
     conf.setClass(MO_PREFIX + namedOutput + KEY, keyClass, Object.class);
     conf.setClass(MO_PREFIX + namedOutput + VALUE, valueClass, Object.class);
-    ConfigurationUtil.setNamedConfigurations(conf, namedOutput, outputConfigs);
+    ConfigurationUtil.setNamedConfigurations(conf, computePrefixName(namedOutput), outputConfigs);
+  }
+
+  private static String computePrefixName(String outputName) {
+    return PREFIXED_CONF_PREFIX + outputName + CONF;
+  }
+
+  private static void assertAlphaNumeric(String name) {
+    Preconditions.checkArgument(StringUtils.isAlphanumeric(name),
```

Comment:
This may be too strict. Since you can use a dataset name or a stream name, we need to allow anything allowed for datasets or streams.